### PR TITLE
Prebid Core: creative comment injection spot reverted

### PR DESCRIPTION
**Pull request by denys-berzoy-confiant**
- reverted injection point of creative comment to pre-PR #6860
- added code to reinject comment in case it was removed upon rendering

## Type of change
- [x] Bugfix

## Description of change
This is an adjustment to the code previously pushed (#6860) to address the issue of disappearing bid detail comments in the document. We (Confiant) as a provider of ad quality and security need to have access to the bid detail comment before rendering starts, to match it against our data. The previous push broke our logic. We would like to put the injection of the comment back to the pre-rendering stage and only re-inject it again if it was in fact removed by the adaptor rendering as noticed by the previous developer (@jsut).

### CODE REVIEW

**Comment by jsut** [comment]
It seems weird to me that you've built code on top of of this functionality. Is hooking BID_WON, or AD_RENDER_SUCCEEDED not sufficient to your use case? That's actually a documented API that people acknowledge others rely on and should be changed as lightly as this was previously.

All that said, all I care about is that the comment is still in the DOM after rendering, no matter how the the ad is rendered, so the code itself /seems/ fine to me at a glance.  I'm just worried that there is nothing in this PR that will make people think twice about changing this in the future, so we might end up right back here again.

**Comment by denys-berzoy-confiant** [comment]
@jsut simulating necessary behaviour here is a bit painful, but I'll add the tests. It just takes some time to do necessary manipulations inside the existing testing framework, since the Document object is just a mock and doesn't do its internal magic for us.

**Review by jsnellbaker — APPROVED** [review]
I think this looks okay.  Needs a 2nd approval before merging.

**Review by mmoschovas — APPROVED** [review]
lgtm

**Pull request merged**
